### PR TITLE
List article

### LINKED ISSUE #33: Create list article page
Create the page (home?) to list articles

Article listing should contain:

1. Title
2. Brief description (few lines of the article content)
3. Author
4. Date
5. Comment count (?)
6. Tags

This will require creating a new 'find' method on the controller and service.

Should also create a filter to search articles containing any of the aforementioned data

### PR DESCRIPTION
**Pull request by Asbelll**
Closes #33 

- Create list method to get information from database with optional filters
- Create GET route for article list
- Create Repositories layer for database logic
- Update Express
<br />

**TODO**
- Create pagination
- Limit body preview on the list page

### CODE REVIEW

**Comment by MtsReis** [comment]
We should remove the field for tags after everything is implemented and come up with a better UI in the future. As of now, all we need is the back end to expect tags in the query.

File: src/resources/views/article/list.pug
Comment (MtsReis) [inline src/resources/views/article/list.pug:1]: We need a better way to get a router URL from pug files.

**Comment by Asbelll** [comment]
We have to come back to this whenever we implement users, missing features due to this are date, comment count and author.

File: src/app.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import express from "express";
 import "express-async-errors";
 
+import { NamedRouter, routes as registeredRoutes } from "reversical";
```
Comment (MtsReis) [inline src/app.js:4]: Added "[reversical](https://www.npmjs.com/package/reversical)" for reverse routing

**Review by MtsReis — APPROVED** [review]
Lgtm

**Pull request merged**